app/components/search/GeoSearch.tsx

```diff
@@ -33,30 +39,79 @@ const GeoSearch = () => {
       }
     }
 
-    if (status === "idle") {
-      if (map?.getLayer("hits")) map.removeLayer("hits");
-      if (map?.getSource("hits")) map.removeSource("hits");
-
-      map?.addSource("hits", {
-        type: "geojson",
-        data: hitsToFeatureCollection(renderState.gca.hits.items),
-      });
-
-      map?.addLayer({
-        id: "hits",
-        type: "symbol",
-        source: "hits",
-        layout: {
-          "icon-image": "pulsing-dot",
-        },
-        filter: ["==", "$type", "Point"],
-      });
-    }
+    if (map.getLayer("hits")) map.removeLayer("hits");
+    if (map.getSource("hits")) map.removeSource("hits");
+
+    map.addSource("hits", {
+      type: "geojson",
+      data: hitsToFeatureCollection(hits),
+    });
+
+    map.addLayer({
+      id: "hits",
+      type: "symbol",
+      source: "hits",
+      layout: {
+        "icon-image": "pulsing-dot",
+      },
+      filter: ["==", "$type", "Point"],
+    });
+
+    const handleUnclusteredPointClick = (e: MapLayerMouseEvent) => {
+      if (!e.features || !e.features.length) return;
+
+      const feature = e.features[0];
+      const clickedHit = hits.find(
+        (hit) => hit[modelFieldUUIDs.identifier] === feature.properties.identifier
+      );
+
+      if (clickedHit) {
+        const clickedPlace: TPlaceRecord = {
+          uuid: clickedHit.uuid,
+          name: clickedHit.name,
+          description: clickedHit.description,
+          place_names: clickedHit.place_names || [],
+          place_layers: clickedHit.place_layers || [],
+          web_identifiers: clickedHit.web_identifiers || [],
+          place_geometry: { geometry_json: clickedHit.geometry } || null,
+          user_defined: clickedHit.user_defined || false,
+          identifier: clickedHit[modelFieldUUIDs.identifier],
+          iiif_manifest: clickedHit.iiif_manifest || null,
+        };
+
+        setActivePlace(clickedPlace);
+      }
+    };
+
+    map.on("click", "hits", handleUnclusteredPointClick);
+
+    map.on("mouseenter", "hits", () => {
+      map.getCanvas().style.cursor = "pointer";
+    });
+    map.on("mouseleave", "hits", () => {
+      map.getCanvas().style.cursor = "";
+    });
 
-    return () => {};
-  }, [renderState, map, mapLoaded, status]);
+    return () => {
+      if (map.getImage("pulsing-dot")) map.removeImage("pulsing-dot");
```

Comment:
This is an important thing to do, however, it seems to fail when navigating to a different route. I suggest wrapping lines 96 - 99 in [try, catch blocks.](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/try...catch)